src/main/java/com/checkmarx/flow/controller/ADOController.java

```diff
@@ -299,6 +302,21 @@
         return getSuccessMessage();
     }
 
+    private Boolean isDeleteBranchEvent(Resource resource){
+        if (resource.getRefUpdates().size() == 1){
+            String newBranchRef = resource.getRefUpdates().get(0).getNewObjectId();
+
+            if (newBranchRef.equals(BRANCH_DELETED_REF)){
+                log.info("new-branch ref is empty - detect ADO DELETE event");
+                return true;
+            }
+            return false;
+        }
+
+        log.info("unexpected number of refUpdates in push event");
```

Comment:
Maybe you can print actual number of refUpdates

Reply:
Added